mmv1/third_party/terraform/services/logging/resource_logging_project_sink_test.go

```diff
@@ -415,6 +455,115 @@ resource "google_storage_bucket" "log-bucket" {
 `, name, envvar.GetTestProjectFromEnv(), bucketName)
 }
 
+func testAccLoggingProjectSink_customWriter(org, billingId, project, name, serviceAccount string) string {
+	return fmt.Sprintf(`
+
+resource "google_project" "destination-project" {
+	project_id      = "%s"
+	name            = "%s"
+	org_id          = "%s"
+	billing_account = "%s"
+}	
+
+resource "google_logging_project_bucket_config" "destination-bucket" {
+    project    = google_project.destination-project.project_id
+    location  = "us-central1"
+    retention_days = 30
+    bucket_id = "shared-bucket"
+}
+
+resource "google_service_account" "test-account1" {
+  account_id   = "%s"
+  display_name = "Log Sink Custom WriterIdentity Testing Account"
+}
+
+resource "google_project_iam_member" "custom-sa-logbucket-binding" {
+  project = google_project.destination-project.project_id
+  role   = "roles/logging.bucketWriter"
+  member = "serviceAccount:${google_service_account.test-account1.email}"
+}
+
+data "google_project" "testing_project" {
+	project_id = "%s"
+}
+
+locals {
+	project_number = data.google_project.testing_project.number
+}
+
+resource "google_service_account_iam_member" "loggingsa-customsa-binding" {
+	service_account_id = google_service_account.test-account1.name
+	role   = "roles/iam.serviceAccountTokenCreator"
+	member = "serviceAccount:service-${local.project_number}@gcp-sa-logging.iam.gserviceaccount.com"
+}
+
+resource "google_logging_project_sink" "custom_writer" {
+  name        = "%s"
+  destination = "logging.googleapis.com/projects/${google_project.destination-project.project_id}/locations/us-central1/buckets/shared-bucket"
+  filter      = "logName=\"projects/%s/logs/compute.googleapis.com%%2Factivity_log\" AND severity>=ERROR"
+
+  unique_writer_identity = true
+  custom_writer_identity = "serviceAccount:${google_service_account.test-account1.email}"
+
+  depends_on = [google_logging_project_bucket_config.destination-bucket]
+}
+`, project, project, org, billingId, serviceAccount, envvar.GetTestProjectFromEnv(), name, envvar.GetTestProjectFromEnv())
+}
+
+func testAccLoggingProjectSink_customWriterUpdated(org, billingId, project, name, serviceAccount string) string {
+	return fmt.Sprintf(`
+	resource "google_project" "destination-project" {
```

Comment:
nit: Do you mind reformatting this config to remove the indentation?

Reply:
ACT.